pkg/devicetwin/dtmanager/membership_test.go

```diff
@@ -0,0 +1,359 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dtmanager
+
+import (
+	"encoding/json"
+	"errors"
+	"sync"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/kubeedge/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/mocks/beego"
+	"github.com/kubeedge/kubeedge/pkg/common/dbm"
+	"github.com/kubeedge/kubeedge/pkg/devicetwin/dtcontext"
+	"github.com/kubeedge/kubeedge/pkg/devicetwin/dttype"
+)
+
+var ormerMock *beego.MockOrmer
+var querySeterMock *beego.MockQuerySeter
+
+func initMocks(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	ormerMock = beego.NewMockOrmer(mockCtrl)
+	querySeterMock = beego.NewMockQuerySeter(mockCtrl)
+	dbm.DBAccess = ormerMock
+}
+
+func TestInitMemActionCallBack(t *testing.T) {
+	initMemActionCallBack()
+}
+
+func TestGetRemoveList(t *testing.T) {
+	dtc := &dtcontext.DTContext{
+		DeviceList: &sync.Map{},
+	}
+
+	var device dttype.Device
+	dtc.DeviceList.Store("DeviceA", "")
+	dtc.DeviceList.Store("DeviceB", &device)
+	dArray := []dttype.Device{}
+	d := dttype.Device{
```

Comment:
Can we have assertions in this function and the function below ? get remove list has a return type

Reply:
Added assert condition for getRemoveList.